samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -129,8 +138,11 @@ JobCoordinator getJobCoordinator() {
     this.jobCoordinator = (jobCoordinator != null) ? jobCoordinator : getJobCoordinator();
     this.jobCoordinatorListener = createJobCoordinatorListener();
     this.jobCoordinator.setListener(jobCoordinatorListener);
+    //this.jcContainerShutdownLatch= new CountDownLatch(1);
```

Comment:
my logic was - we check the latch only when container is not null, and container becomes not null only in onNewJobModel, is it correct?